www-apps/cgit/cgit-1.2.3_p20240802-r1.ebuild

```diff
@@ -0,0 +1,128 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LUA_COMPAT=( lua5-{1..4} luajit )
+
+PYTHON_COMPAT=( python3_{10..14} )
+
+WEBAPP_MANUAL_SLOT="yes"
+
+inherit flag-o-matic lua-single python-single-r1 tmpfiles toolchain-funcs webapp
+
+[[ -z "${CGIT_CACHEDIR}" ]] && CGIT_CACHEDIR="/var/cache/${PN}/"
+
+GIT_V="2.46.0"
+MY_COMMIT="09d24d7cd0b7e85633f2f43808b12871bb209d69"
+
+DESCRIPTION="A fast web-interface for Git repositories"
+HOMEPAGE="https://git.zx2c4.com/cgit/about/"
+SRC_URI="https://www.kernel.org/pub/software/scm/git/git-${GIT_V}.tar.xz
+	https://git.zx2c4.com/cgit/snapshot/cgit-${MY_COMMIT}.tar.xz"
+S="${WORKDIR}/${PN}-${MY_COMMIT}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~riscv ~x86"
+
+IUSE="doc +highlight +lua test"
+REQUIRED_USE="lua? ( ${LUA_REQUIRED_USE} ) ${PYTHON_REQUIRED_USE}"
+
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	acct-group/cgit
+	acct-user/cgit
+	dev-libs/openssl:0=
+	dev-vcs/git
+	highlight? (
+		$(python_gen_cond_dep '
+			dev-python/docutils[${PYTHON_USEDEP}]
+			dev-python/markdown[${PYTHON_USEDEP}]
+			dev-python/pygments[${PYTHON_USEDEP}]
+		')
+		sys-apps/groff
+	)
+	lua? ( ${LUA_DEPS} )
+	sys-libs/zlib
+	virtual/httpd-cgi
+"
+# ebuilds without WEBAPP_MANUAL_SLOT="yes" are broken
+DEPEND="${RDEPEND}"
+BDEPEND="
+	doc? (
+		app-text/docbook-xsl-stylesheets
+		>=app-text/asciidoc-8.5.1
+	)
+"
+
+pkg_setup() {
+	python_setup
+	webapp_pkg_setup
+	use lua && lua-single_pkg_setup
+}
+
+src_configure() {
+	if ! [[ ${PV} =~ 9999* ]]; then
+		rmdir git || die
+		mv "${WORKDIR}"/git-"${GIT_V}" git || die
+	fi
+
+	# bug #951555
+	append-cflags -std=gnu17
+
+	echo "prefix = ${EPREFIX}/usr" >> cgit.conf || die "echo prefix failed"
+	echo "libdir = ${EPREFIX}/usr/$(get_libdir)" >> cgit.conf || die "echo libdir failed"
+	echo "CGIT_SCRIPT_PATH = ${MY_CGIBINDIR}" >> cgit.conf || die "echo CGIT_SCRIPT_PATH failed"
+	echo "CGIT_DATA_PATH = ${MY_HTDOCSDIR}" >> cgit.conf || die "echo CGIT_DATA_PATH failed"
+	echo "CACHE_ROOT = ${CGIT_CACHEDIR}" >> cgit.conf || die "echo CACHE_ROOT failed"
+	if use lua; then
+		echo "LUA_PKGCONFIG = ${ELUA}" >> cgit.conf || die "echo LUA_PKGCONFIG failed"
+	else
+		echo "NO_LUA = 1" >> cgit.conf || die "echo NO_LUA failed"
+	fi
+}
+
+src_compile() {
+	emake V=1 AR="$(tc-getAR)" CC="$(tc-getCC)" CFLAGS="${CFLAGS}" LDFLAGS="${LDFLAGS}"
+	use doc && emake V=1 doc-man
+}
+
+src_install() {
+	webapp_src_preinst
+
+	emake V=1 AR="$(tc-getAR)" CC="$(tc-getCC)" CFLAGS="${CFLAGS}" LDFLAGS="${LDFLAGS}" DESTDIR="${D}" install
+
+	insinto /etc
+	doins "${FILESDIR}"/cgitrc
+
+	dodoc README
+	use doc && doman cgitrc.5
```

Comment:
This bit breaks a [QA policy](https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0305).  I will not hold merging it for now, as it's not new, but we should add it to a TODO list and fix it eventually.